### PR TITLE
Match CLR data types to MySQL. fixes #75

### LINKED ISSUE #75: Align CLR and MySQL Data Types
Right now we have a few issues where the MySQL Data Type we are using is not the best type for the CLR DataType.

	CLR Type      Current MySQL Type    Desired MySQL Type
	-------------------------------------------------------------------------------
	ushort        int                   smallint unsigned
	uint          bigint                int unsigned
	ulong         real(20, 0)           bigint unsigned
	char          int                   tinyint
	decimal       decimal(18, 2)        decimal(65, 30)

Additionaly, the `DateTimeOffset` CLR Type does not map to a MySQL type.  The closest we can get is storing a `long` using `DateTimeOffset.ToUnixTimeMilliseconds` and `DateTimeOffset.FromUnixTimeMilliseconds`

I will open a PR that implements the desired mappings and their associated tests.

### PR DESCRIPTION
**Pull request by caleblloyd**
See #75 for more information

### CODE REVIEW

File: src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlTypeMapper.cs [hunk 1]
```diff
@@ -14,37 +14,50 @@ namespace Microsoft.EntityFrameworkCore.Storage.Internal
...
+	    private readonly RelationalTypeMapping _time             = new RelationalTypeMapping("time(6)", typeof(TimeSpan), DbType.Time);
+
+	    // json
+	    private readonly RelationalTypeMapping _json             = new RelationalTypeMapping("json", typeof(JsonObject<>), DbType.String);
+
+	    // row version
+	    private readonly RelationalTypeMapping _rowversion       = new RelationalTypeMapping("TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP", typeof(byte[]), DbType.Binary);
+
+	    // guid
+	    private readonly RelationalTypeMapping _uniqueidentifier = new RelationalTypeMapping("char(38)", typeof(Guid));
```
Comment (caleblloyd) [inline src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlTypeMapper.cs]: done

File: src/Pomelo.EntityFrameworkCore.MySql/Storage/Utilities.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore.Storage
+{
+	public static class Utilities
+	{
+		#if NET451
```
Comment (caleblloyd) [inline src/Pomelo.EntityFrameworkCore.MySql/Storage/Utilities.cs]: NET451 needs `ToUnixTimeMilliseconds` and `FromUnixTimeMilliseconds`

**Comment by caleblloyd** [comment]
@Kagamine I hit merge early on accident.  Any other changes you see that need to be done here?

**Comment by Kagamine** [comment]
@caleblloyd Are you sure use int instead of datetimeoffset is right? a datetimeoffset value is consist of 2 parts: DateTime and TimeSpan.

**Comment by caleblloyd** [comment]
@Kagamine opened #79 for discussion

**Pull request merged**